.github/workflows/validate-mdx.links.yaml

```diff
@@ -0,0 +1,26 @@
+on:
+  pull_request:
+    paths:
+      # we run to check if the links are wrong
+      - 'packages/web/docs/src/**/*.mdx'
+      # or if a page disappeared
+      - 'packages/web/docs/src/app/**'
+
+jobs:
+  validate-mdx-links:
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 2
+
+      - name: setup environment
+        uses: ./.github/actions/setup
+        with:
+          actor: validate-mdx-links
+          cacheTurbo: false
+
+      - name: Validate Docs MDX Links
+        run: ./scripts/validate-mdx-links.ts --cwd packages/web/docs
```

Comment:
Not sure if this should be separate workflow or part of `lint`. I'll leave this decision to someone else.